dhcpv4/option_subnet_mask.go

```diff
@@ -0,0 +1,56 @@
+package dhcpv4
+
+import (
+	"fmt"
+	"net"
+)
+
+// This option implements the subnet mask option
+// https://tools.ietf.org/html/rfc2132
+
+// OptSubnetMask represents an option encapsulating the subnet mask.
+type OptSubnetMask struct {
+	SubnetMask net.IPMask
+}
+
+// ParseOptSubnetMask returns a new OptSubnetMask from a byte
+// stream, or error if any.
+func ParseOptSubnetMask(data []byte) (*OptSubnetMask, error) {
+	if len(data) < 2 {
+		return nil, ErrShortByteStream
+	}
+	code := OptionCode(data[0])
+	if code != OptionSubnetMask {
+		return nil, fmt.Errorf("expected code %v, got %v", OptionSubnetMask, code)
+	}
+	length := int(data[1])
+	if length != 4 {
+		return nil, fmt.Errorf("unexepcted length: expected 4, got %v", length)
+	}
+	if len(data) < 6 {
+		return nil, ErrShortByteStream
+	}
+	return &OptSubnetMask{SubnetMask: net.IPMask(data[2 : 2+length])}, nil
+}
+
+// Code returns the option code.
+func (o *OptSubnetMask) Code() OptionCode {
+	return OptionSubnetMask
+}
+
+// ToBytes returns a serialized stream of bytes for this option.
+func (o *OptSubnetMask) ToBytes() []byte {
+	ret := []byte{byte(o.Code()), byte(o.Length())}
+	return append(ret, o.SubnetMask[:4]...)
```

Comment:
nit: this should be `.To4()` instead of `[:4]`. It's a theoretical issue here, because we expect this to be set by ParseOptSubnetMask, but more in general this only works correctlyif len(o.SubnetMask) is 4